proposals/nnnn-cloud-app-transport-adapter.md

```diff
@@ -0,0 +1,243 @@
+  CLOUD,
+  AOA,
+  PASA_AOA,
+  BLUETOOTH,
+  PASA_BLUETOOTH,
+  MME,
+  IOS_BT,
+  IOS_USB,
+  TCP,
+  UNKNOWN
+};
+```
+
+#### Websocket Client Transport Adapter
+Core will need a new transport adapter that initiates websocket client connections to a remote endpoint. I suggest creating a transport adapter that integrates this library: [Boost Beast](https://github.com/boostorg/beast) for handling the websocket connections.  
+
+### Policies
+
+app_policies section of policy table related classes must be expanded to support new optional endpoint and ssl_certificate fields. 
+
+types.h
+```
+struct PolicyBase : CompositeType {
+ public:
+  Strings groups;
+  Optional<Strings> preconsented_groups;
+  Enum<Priority> priority;
+  Enum<HmiLevel> default_hmi;
+  Boolean keep_context;
+  Boolean steal_focus;
+  Optional<String<0, 255> > cloud_app_endpoint;
+  Optional<String<0, 65535> > ssl_certificate;
+```
+
+policy_table_interface_ext.xml
+```
+    <struct name="ApplicationParams">
+        <param name="groups" type="Strings" />
+        <param name="nicknames" type="Strings" mandatory="false" />
+        <param name="preconsented_groups" type="Strings" mandatory="false" />
+        <param name="AppHMIType" type="AppHMITypes" mandatory="false" />
+        <param name="priority" type="Priority" />
+        <param name="default_hmi" type="HmiLevel" />
+        <param name="keep_context" type="Boolean" />
+        <param name="steal_focus" type="Boolean" />
+        <param name="memory_kb" type="Integer" minvalue="1" maxvalue="65225" mandatory="false"/>
+        <param name="watchdog_timer_ms" type="Integer" minvalue="1"
+            maxvalue="65225" mandatory="false"/>
+        <param name="cloud_app_endpoint", type="String", minlength="0" maxlength="255", mandatory="false">
+        <param name="certificate" type="String" minlength="0" maxlength="255"
+            mandatory="false" />
+    </struct>
+```
+This feature should be supported by both regular and external policy build configurations.
+
+### SDL Server
+
+SDL Server must create an interface for managing additional policy data. Interface should allow for adding, updating, and revoking endpoint urls and SSL certificates. This work should be minimal since its only two extra parameters.
+
+## Alternatives considered
+
+Since there are many aspects to this proposal, I have considered a few alternatives.
+
+### Using Tokens to authenticate
+
+1. Core gets web app endpoints and associated tokens from policy server after a policy table update. 
+2. Core opens a websocket connection for each webapp endpoint located in the policy table. 
+3. Core sends the VIN and token to the application and waits for the application to respond with data.
+4. The web application server sends a request to the policy server to identify the incoming connection via the received VIN and token. 
+5. If the policy server responds matches that the VIN and token are valid it responds with success to the web app server. 
```

Comment:
This needs to be re-worded.  Maybe "If the policy server response matches the VIN, and the token is valid, the policy server responds with success..."  I'm not completely sure what the intended information is here.